### PR TITLE
Make sure One Shot Optimizer always returns the same suggestion

### LINKED ISSUE #755: mlos_bench: Benchmarking mode
Benchmarking mode will suggest a config and then forget about the default config it's supposed to run with.

1. OneShotOptimizer will change the default config
2. Default config is lost, unless is user manually creates one and passes is it with --tunable-values 

**_Should be automatically running with defaults if not opt mentioned in cli_**

### PR DESCRIPTION
**Pull request by motus**
Make sure the `OneShotOptimizer.suggest()` call always returns the same configuration.

Changes related to `MockEnv` determinism moved to #769 

Closes #755

### CODE REVIEW

File: mlos_bench/mlos_bench/config/schemas/environments/mock-env-subschema.json [hunk 1]
```diff
@@ -19,12 +19,12 @@
                 },
                 {
                     "properties": {
-                        "seed": {
+                        "mock_env_seed": {
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/config/schemas/environments/mock-env-subschema.json:22]: Not sure I follow the logic of changing this particular one.  Why not let that draw from other global `seed` environment variables as well?

Comment (bpkroth) [reply mlos_bench/mlos_bench/config/schemas/environments/mock-env-subschema.json:22]: Also, the `MockEnv` changes might be better in a different PR

File: mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py [hunk 2]
```diff
@@ -33,6 +33,14 @@ def __init__(self,
         _LOG.info("Run a single iteration for: %s", self._tunables)
         self._max_iter = 1  # Always run for just one iteration.
 
+    def suggest(self) -> TunableGroups:
+        """
+        Always produce the same (initial) suggestion.
+        """
+        tunables = super().suggest()
+        self._start_with_defaults = True
```
[the remark this answers is not in the document]
Comment (bpkroth) [reply mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py:41]: Do we have a test for those two modes?

File: mlos_bench/mlos_bench/tests/launcher_in_process_test.py [hunk 3]
```diff
@@ -17,11 +17,14 @@
     ("argv", "expected_score"), [
         ([
             "--config", "mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc",
-        ], 65.6742),
+            "--trial_config_repeat_count", "5",
+            "--mock_env_seed", "0",  # Deterministic Mock Environment.
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/launcher_in_process_test.py:21]: Ahh!  That's nice.

**Review by bpkroth — CHANGES_REQUESTED** [review]
Let's split the MockEnv changes out for clarity.
Then we can add some more tests for the new (expected) OneShot behavior in this PR.

File: mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (motus) [reply mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py:41]: > Does this work when we have `--tunable-values` passed?
it does - we set the tunable values in the launcher before passing the tunables to the optimizer's constructor

File: mlos_bench/mlos_bench/config/schemas/environments/mock-env-subschema.json (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/config/schemas/environments/mock-env-subschema.json:22]: > Why not let that draw from other global seed environment variables as well?

Our `seed` parameter is also used to switch on the deterministic behavior. I want to separate the determinism in the optimizer and mock environment. Also, global parameters with short names like `seed` or `metrics` are prone to name collisions and our parameter substitution behavior makes it hard to debug. So yeah, naming things is hard 😺

File: mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (motus) [reply mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py:41]: added a unit test to this PR.

**Comment by bpkroth** [comment]
Split some changes out to #769

File: mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bpkroth) [reply mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py:41]: The new test from #769 checks the MockOptimizer, right?  Do we have one that checks OneShot specifically?  What about if it's called repeatedly in separate invocations (e.g., like the situation @yshady) was working on?  Can we add tests for that?

[the remark this answers is not in the document]
Comment (bpkroth) [reply mlos_bench/mlos_bench/optimizers/one_shot_optimizer.py:41]: Am specifically wondering about the case for repeated calls across process invocations, but let's take that up in a future PR.

**Review by bpkroth — APPROVED** [review]

**Pull request merged**